### PR TITLE
Show captions in gallery page view

### PR DESCRIPTION
**Pull request by michaelkirk-signal**
PTAL @charlesmchen

### CODE REVIEW

File: Signal/src/ViewControllers/MediaGalleryViewController.swift [hunk 1]
```diff
@@ -31,6 +31,10 @@ public class MediaGalleryItem: Equatable, Hashable {
         return attachmentStream.isImage
     }
 
+    var caption: String? {
+        return attachmentStream.caption
```
Comment (charlesmchen-signal) [inline Signal/src/ViewControllers/MediaGalleryViewController.swift]: We need to apply `filterStringForDisplay`.  Maybe we should add `[TSAttachment captionForDisplay]`?

File: Signal/src/ViewControllers/MediaPageViewController.swift [hunk 2]
```diff
@@ -654,3 +755,38 @@ class MediaPageViewController: UIPageViewController, UIPageViewControllerDataSou
...
+        addSubview(gradientView)
+        gradientView.autoPinEdgesToSuperviewEdges()
+
+        addSubview(label)
+        label.font = UIFont.ows_dynamicTypeBody
+        label.textColor = .white
+
+        // Usually captions are short, but they can be as long as 2k.
+        // We don't have UI for viewing infinitely large captions, so
+        // we do some not-ideal things to broaden the lenght of the
```
Comment (charlesmchen-signal) [inline Signal/src/ViewControllers/MediaPageViewController.swift]: s/lenght/length/.

`minimumScaleFactor` a) reduces accessibility by rendering the text smaller, making it harder to read. b) still doesn't guarantee us that all text will fit in a reasonable way.  

What if instead we set `numberOfLines` to something like 5 and if users tap on the caption we push a simple view to display all of the text, e.g. `LongTextViewController`.  LGTV isn't pretty but it ensures that all of the caption is readable by all.

Comment (michaelkirk-signal) [reply Signal/src/ViewControllers/MediaPageViewController.swift]: Good point. Long captions are now in a scroll view. Getting a scroll view to grow to fit it's content, but only up to a point, was a little tricky.

File: Signal/src/ViewControllers/MediaGalleryViewController.swift [hunk 3]
```diff
@@ -4,6 +4,12 @@
 
 import Foundation
 
+extension String {
+    var filterForDisplay: String? {
```
Comment (michaelkirk-signal) [inline Signal/src/ViewControllers/MediaGalleryViewController.swift]: whoops, I'm going to move this to Strings+OWS.swift

File: Signal/src/ViewControllers/MediaPageViewController.swift [hunk 4]
```diff
@@ -654,3 +750,102 @@ class MediaPageViewController: UIPageViewController, UIPageViewControllerDataSou
...
+            } else if isScrollEnabled {
+                isScrollEnabled = false
+            }
+        }
+
+        override var intrinsicContentSize: CGSize {
+            var size = super.intrinsicContentSize
+
+            if size.height == UIViewNoIntrinsicMetric {
+                size.height = layoutManager.usedRect(for: textContainer).height + textContainerInset.top + textContainerInset.bottom
```
Comment (charlesmchen-signal) [inline Signal/src/ViewControllers/MediaPageViewController.swift]: TIL

**Comment by michaelkirk-signal** [comment]
in cc068b0b7

**Pull request closed without merging**